## 作废相关
1. 只有发起人才可以作废。
2. 只有作废的才可以删除。
```
 public ActionResult WFEnd(string endReason, string busId)
        {
            string reason = Server.UrlDecode(endReason);
            M_Return m_Return = new M_Return();
            try
            {
                if (!string.IsNullOrWhiteSpace(busId) && busId != Guid.Empty.ToString())
                {
                    PER_WORKERSNEED model = PER_WORKERSNEEDBll.Instance.LoadID(new Guid(busId));
                    if (WFWCFFactory.Instance.WFServer.StopWorkFlow(model.NeedId.ToString()))
                    {
                        model.CheckStatus = (int)Enumerator.CheckStatus.作废;
                        PER_WORKERSNEEDBll.Instance.Update(model);
                        WFWCFFactory.Instance.WFServer.ModifyWorkFlowStatus(model.NeedId.ToString(), ((int)Enumerator.EnumBillState.End).ToString());
                        m_Return.Value = true;
                        m_Return.Info = "成功:流程中止成功!";
                        return Content(m_Return.Info);
                    }
                    else
                    {
                        m_Return.Value = true;
                        m_Return.Info = "失败:流程中止失败!";
                        return Content(m_Return.Info);
                    }
                }
                else
                {
                    m_Return.Value = true;
                    m_Return.Info = "失败:流程中止失败!";
                    return Content(m_Return.Info);
                }
            }
            catch (Exception ex)
            {
                return Content(m_Return.Info + ex.Message);
            }
```

## 关于删除 
1. 只删除业务表数据，不关心流程表数据。
```
public ActionResult WorkersneedAskDelete(string NeedID)
        {
            try
            {
                string sql = "delete PER_WORKERSNEED where NeedID='" + NeedID + "';";
                sql += "delete PER_POST_ASK where NeedID='" + NeedID + "';";
                PER_WORKERSNEEDBll.Instance.LoadListSelect(sql);
                return Content("0,/Recruit/PerWorkersneedIndex");
            }
            catch (Exception ex)
            {
                return Content("1,删除失败！");
            }
        }
```

